esp-wifi/src/lib.rs

```diff
@@ -181,10 +186,29 @@ impl EspWifiInitialization {
     }
 }
 
+#[derive(Debug, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[non_exhaustive]
+/// An internal struct designed to make [`EspWifiDeinitialization`] uncreatable
+/// outside of this crate.
+pub struct EspWifiDeinitializationInternal;
```

Comment:
We shouldn't need multiple internal-but-externally-not-constructible types

Reply:
Totally nuked, design has been changed several times and they are no longer needed.